/*
  Function Calls

  This API Section holds messages used for executing remotely triggerable functions following a generic schema like it is defined in the GENIVI Vehicle Service Catalog (VSC) specification.  While there already is a \"Remote Functions\" service in the MC, that solution is based on the proprietary Grade-X technology, which is not open to non-Grade-X devices. Also Grade-X is rather focused on remote diagnostic use cases, while an OFC feature designed for remotely triggering arbitrary functions could leverage semantics of the contained functions better.

  The version of the OpenAPI document: 2

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package com.bosch.bfb.protobuf.functioncalls.v2;

import public "models/device_to_backend_message.proto";
import public "models/device_to_backend_message_meta.proto";
import public "models/function_call_status_event_all_of.proto";

message FunctionCallStatusEvent {

  DeviceToBackendMessageMeta meta = 3347973;

  // Current status of a previously triggered/running/pending function call
  enum StatusEnum {
    UNKNOWN = 0;
    PENDING = 1;
    SUCCESS = 2;
    FAILURE = 3;
  }

  StatusEnum status = 355610639;

  // Generic output argument. Need to be defined as JSON string (e.g. \"\"{\"seat\":{\"location\":{\"row\":0,\"index\":0},\"position\":{\"base\":500,\"cushion\":500,\"lumbar\":0,\"side_bolster\":0,\"head_restraint\":0}}}\"\")
  string outArguments = 301089192;

}
